src/Components/Common/SearchBox/SearchBox.jsx

```diff
@@ -28,6 +37,55 @@ class SearchBox extends Component {
     this.setState({ filter: filterValue });
   };
 
+  /**
+   * submit the search filter
+   * @param {object} e
+   * @returns {void}
+   */
+  onSubmitSearch = e => {
+    e.preventDefault();
+    let url;
+    const { value } = this.props;
+    const { filter } = this.state;
+    const { onSearch } = this.props;
+    const { history } = this.props;
+
+    if (value) {
+      switch (filter) {
+        case 'author':
+          url = `/api/search?author=${value}`;
+          break;
+        case 'title':
+          url = `/api/search?title=${value}`;
+          break;
+        case 'tag':
+          url = `/api/search?tag=${value}`;
+          break;
+        default:
+          url = `/api/search?keyword=${value}`;
+      }
```

Comment:
@coolbeatz71 This code can be written simply like this : `/api/search?{filter || 'keyword'}=${value}`

Reply:
Wow, @gracelungu good idea brother. I am refactoring now. Thx